src/libos/src/net/socket/unix/stream/stream.rs

```diff
@@ -228,12 +228,17 @@ impl Stream {
         if !flags.is_empty() {
             warn!("unsupported flags: {:?}", flags);
         }
-        if msg_hdr.get_control().is_some() {
-            warn!("sendmsg with msg_control is not supported");
-        }
 
         let bufs = msg_hdr.get_iovs().as_slices();
-        self.writev(bufs)
+        let mut data_len = self.writev(bufs)?;
+
+        if let Some(msg_control) = msg_hdr.get_control() {
+            self.set_ancillary(Ancillary {
```

Comment:
I think it would be better to call `set_ancillary` in stream's `connect` and `accept`.